### ISSUE TITLE
receiving incomplete data from PX4Flow to PX4 fmu-v5

### ISSUE THREAD

**Issue by 60ericleal60** [PX4/Firmware#11227]
I forked the PX4 repository on Jan 10 2019 to add some air frames and uploaded that to my Pixhawk 4. I cloned the PX4Flow repository and uploaded it to the optical flow sensor today the 16th. When the PX4Flow board is connected to my computer over USB to QGC all looks good but when it is going through the Pixhawk through the I2C bus I only get some of the data with mostly 0's

![screenshot from 2019-01-16 17-34-31](https://user-images.githubusercontent.com/38165926/51283574-a84ef580-19b6-11e9-92ba-606ea09e4fcd.png)

I then hooked the PX4Flow board to a Nucleo f767zi and read the outputs with a simple read and display of the 26 byte register 0x16 and all of the data came through correctly. Any ideas of whats going on and where I should look?

**Comment by mhkabir** [comment]
Can you try unplugging and replugging the flow module without rebooting the pixhawk and see if the data starts coming through?

**Comment by 60ericleal60** [comment]
Ok so when I do that I will get 3 or 4 readings in a row and then it goes back to zeros.

**Comment by 60ericleal60** [comment]
I uploaded the Ardupilot firmware to the PX4Flow board from QGroundControl to see if the readings to the NUCLEO board were any different from the 0x16 register and it was not. However when I connected the PX4Flow board with the Ardupilot firware to the Pixhawk with the PX4 stack, all of the data comes through to the MAVlink inspector. I also tried putting the stable release of PX4 1.8.2 on the Pixhawk and optical flow does not show up in the Mavlink inspector at all.
